src/app/Http/Controllers/AccountController.php

```diff
@@ -348,4 +353,57 @@ public function updateMail(Request $request)
         }
         return redirect('/');
     }
+
+    public function pdfTicket(string $ticketId): Response {
```

Comment:
This function does not belong here. It is accessed in the account and in the event (via the partial), but it clearly belongs to the EventParticipant. So this function should be moved to the EventParticipant itself and then take advantage of the router automatic type assignment . `ticketId` is also wrong here, it is eventParticipantId (as you also see in your query). The query can be removed completeley, since the router takes care of that, if done right.